DERMS/FieldProcessor/BlockingQueue.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace FieldProcessor
+{
+    public class BlockingQueue<T> : IDisposable
```

Comment:
Da li je razmotreno koriscenje `BlockingCollection<T>`? Njegova defaultna kolekcija je `ConcurrentQueue<T>` tako da mi deluje da nema potrebe za ovim.

Ako nesto nisam uzeo u obzir slobodno me ispravi.

Jeste da se vec na dosta mesta koristi u ovom PR-u, ali radije bi da koristimo nesto sto je vec napisano

Reply:
Razlika je u tome sto ConcurrentQueue nije blokirajuci, ideja je bila da operacija dequeue bude blokirajuca, ukoliko je queue prazan, onaj ko vrsi Dequeue "otkocice se" nakon punjena kolekcije. Hteo sam da zaobidjem upotrebu Thread.Sleep ili non stop TryDequeue.

Moguce je realizovati to resenje i sa ConcurrentQueue ali opet cu imati neku Wrapper klasu za blokirajucu funkcionalnost.

Mislim da je bitno sacuvati procesorkse resurce na SCADA delu, otuda i ovo resenje.

Reply:
Cini mi se da se nismo razumeli. Predlog je bio da se koristi `BlockingCollection<T>` cija je defaultna kolekcija ispod haube `ConcurrentQueue<T>`. Drugim recima to je bas blokirajuci queue koji je vec implementiran.

https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.blockingcollection-1?view=netframework-4.8#remarks

Reply:
Nismo se bili razumeli, izgleda sam izmisljao toplu vodu. Napravicu issue za ovo. Lepo si ovo ucio 💯